### PR TITLE
Support Auto-scaling status

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
Add status for auto-scaling

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Add `status` field for `TidbAutoScaler` CR
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go [hunk 1]
```diff
@@ -149,6 +149,49 @@ type TidbMonitorRef struct {
...
+type TikvAutoScalerStatus struct {
+	BasicAutoScalerStatus `json:",inline"`
+}
+
+// +k8s:openapi-gen=true
+// BasicAutoScalerStatus describe the basic auto-scaling status
+type BasicAutoScalerStatus struct {
+	// MetricsStatusList describes the metrics status in the last auto-scaling reconciliation
+	// +optional
+	MetricsStatusList []MetricsStatus `json:"metrics,omitempty"`
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go:180]: Add `lastScaleTime` as HPA [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#horizontalpodautoscalerstatus-v1-autoscaling)?

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go:180]: updated

File: pkg/autoscaler/autoscaler/autoscaler_manager.go [hunk 2]
```diff
@@ -125,9 +128,44 @@ func (am *autoScalerManager) syncTidbClusterReplicas(tac *v1alpha1.TidbClusterAu
...
+		return "", nil
+	}
+
+	if tac.Spec.TiKV != nil {
+		tac.Status.TiKV.CurrentReplicas = oldTc.Status.TiKV.StatefulSet.CurrentReplicas
+		lastTimestamp, err := f(label.AnnTiKVLastAutoScalingTimestamp)
+		if err != nil {
+			return err
+		}
+		tac.Status.TiKV.LastAutoScalingTimestamp = &lastTimestamp
```
Comment (DanielZhangQD) [inline pkg/autoscaler/autoscaler/autoscaler_manager.go:155]: If `lastTimeStamp==""`, do we still need to set it here?

Comment (Yisaer) [reply pkg/autoscaler/autoscaler/autoscaler_manager.go]: updated.

**Comment by Yisaer** [comment]
@weekface @cofyc  PTAL

File: pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go [hunk 3]
```diff
@@ -149,6 +149,52 @@ type TidbMonitorRef struct {
...
+	// CurrentReplicas describes the current replicas for the component(tidb/tikv)
+	CurrentReplicas int32 `json:"currentReplicas"`
+	// RecommendedReplicas describes the calculated replicas in the last auto-scaling reconciliation for the component(tidb/tikv)
+	// +optional
+	RecommendedReplicas *int32 `json:"recommendedReplicas,omitempty"`
+	// LastAutoScalingTimestamp describes the last auto-scaling timestamp for the component(tidb/tikv)
+	// +optional
+	LastAutoScalingTimestamp *string `json:"lastAutoScalingTimestamp,omitempty"`
```
Comment (weekface) [inline pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go:188]: why use `*string` as timestamp?

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go:188]: Do you mean to use `string` for timestamp or other struct?

Comment (weekface) [reply pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go:188]: Can you use `*time.Time`?

Comment (cofyc) [reply pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go:188]: yes, then we don't need to parse it if we want to call time functions on it. by the way, it's better to use `*metav1.Time` in Kubernetes. It's a wrapper around `time.Time` which supports correct marshaling to YAML and JSON.

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go]: good advice, updated.

**Comment by cofyc** [comment]
@Yisaer pick into release-1.1?

**Comment by Yisaer** [comment]
@cofyc  sure, I would do it manually.

**Pull request merged**